src/Bounds.cpp

```diff
@@ -547,11 +548,14 @@ class Bounds : public IRVisitor {
 
     void visit(const Load *op) {
         op->index.accept(this);
-        if (interval.is_single_point()) {
-            // If the index is const we can return the load of that index
+        if (interval.is_single_point() && is_one(op->predicate)) {
+            // If the index is const and it is not a predicated load,
```

Comment:
I'm still a bit unclear on this. Eliminate bool vectors shouldn't be changing a scalar predicated load into a mask predicate, so why generate one here? Scalar load predicates don't need to be masks?